launcher/src/main/java/org/apache/spark/launcher/FilteredObjectInputStream.java

```diff
@@ -42,12 +42,20 @@ class FilteredObjectInputStream extends ObjectInputStream {
   protected Class<?> resolveClass(ObjectStreamClass desc)
       throws IOException, ClassNotFoundException {
 
-    boolean isValid = ALLOWED_PACKAGES.stream().anyMatch(p -> desc.getName().startsWith(p));
+    boolean isValid = isValid(desc);
     if (!isValid) {
       throw new IllegalArgumentException(
         String.format("Unexpected class in stream: %s", desc.getName()));
     }
     return super.resolveClass(desc);
   }
 
+  private static boolean isValid(ObjectStreamClass desc) {
+    for (String p : ALLOWED_PACKAGES) {
+      if (desc.getName().startsWith(p)) {
+        return true;
+      }
+    }
+    return false;
+  }
```

Comment:
Given this is not a perf critical path, I would recommend to keep this as is.

Reply:
ok